packages/x-components/src/x-modules/history-queries/components/__tests__/history-queries-switch.spec.ts

```diff
@@ -52,8 +57,31 @@ describe('testing HistoryQueriesSwitch component', () => {
 
     expect(disableListener).toHaveBeenCalledTimes(1);
   });
+
+  it('should emit confirm disable event if there are not history queries', () => {
+    const { wrapper } = renderHistoryQueriesSwitch({
+      historyQueries: [],
+      isEnabled: true
+    });
+    const listener = jest.fn();
+    wrapper.vm.$x.on('UserClickedConfirmDisableHistoryQueries').subscribe(listener);
+
+    wrapper.trigger('click');
+
+    expect(listener).toHaveBeenCalledTimes(1);
```

Comment:
Can you also check the `isEnabled` flag in the `store` `state`?

```suggestion
  await new Promise(resolve => setTimeout(resolve));

  expect(wrapper.vm.$store.state.x.historyQueries.isEnabled).toBe(false);
  expect(listener).toHaveBeenCalledTimes(1);
```